Triangulation_3/include/CGAL/Triangulation_3.h

```diff
@@ -1673,13 +1673,32 @@ class Triangulation_3
   template < class VertexRemover >
   void fill_hole_2D(std::list<Edge_2D>& hole, VertexRemover& remover);
 
-  void make_hole_3D(Vertex_handle v,
-                    Vertex_triple_Facet_map& outer_map,
-                    std::vector<Cell_handle>& hole);
-  // When the incident cells are already known
-  void make_hole_3D(Vertex_handle v,
-                    const std::vector<Cell_handle>& incident_cells,
-                    Vertex_triple_Facet_map& outer_map);
+  Vertex_triple_Facet_map
+  create_hole_outer_map(Vertex_handle v,
+                        const std::vector<Cell_handle>& hole);
+
+  template < class Triangulation >
+  static Vertex_triple_Facet_map
+  create_triangulation_inner_map(const Triangulation& t,
+                                 const Vertex_handle_unique_hash_map& vmap,
+                                 bool inf);
```

Comment:
I always thought it was a bad idea to return a map, but you probably will tell me that there happens a move on return.

Reply:
The worst case would be move, but the usual case is the [Return value optimization](https://sigcpp.github.io/2020/06/08/return-value-optimization): the object is created directly in the stack of the caller instead of the stack of the callee. That optimization is mandatory since C++17, but has been a common optimization of C++ compilers even since the age of C++03.